### ISSUE TITLE
Passed FeaturizationConfig is ignored when in a pipeline AutoMLStep

### ISSUE THREAD

**Issue by kelmahrsi** [Azure/MachineLearningNotebooks#1077]
I have an Azure ML pipeline that contains an Auto ML step. The step involves custom featurization, which consists in dropping unwanted columns. It is created by a function that looks like this:

```
def create_automl_step(training_data, validation_data, model_data, metrics_data, compute_target):
    featurization_config = FeaturizationConfig()
    featurization_config.add_column_purpose("DATETIME", "DateTime")
    featurization_config.add_drop_columns(<list_of_columns_to_drop>)

    automl_config = AutoMLConfig(
        task="regression",
        label_column_name=<target_column>,
        training_data=training_data,
        validation_data=validation_data,
        compute_target=compute_target,
        primary_metric="r2_score",
        featurization=featurization_config,
        ...
    )

    return AutoMLStep(
        name="automl_train",
        automl_config=automl_config,
        inputs=[training_data, validation_data],
        outputs=[metrics_data, model_data],
        allow_reuse=True,
    )
```

A separate code snippet uses the function to create the step and integrate it into a pipeline. It goes something like this:

```
train = PipelineData("train", datastore).as_dataset()
validation = PipelineData("validation", datastore).as_dataset()

# both train and validation are produced by a PythonScriptStep

metrics_output_name = "metrics_output"
best_model_output_name = "best_model_output"

metrics_data = PipelineData(
    name="metrics_data",
        datastore=datastore,
        pipeline_output_name=metrics_output_name,
        training_output=TrainingOutput(type="Metrics"),
    )
    model_data = PipelineData(
    name="model_data",
    datastore=datastore,
    pipeline_output_name=best_model_output_name,
    training_output=TrainingOutput(type="Model"),
)

automl_step = create_automl_step(
    training_data=train.parse_parquet_files(),
    validation_data=validation.parse_parquet_files(),
    model_data=model_data,
    metrics_data=metrics_data,
    compute_target=aml_compute,
)

model_training_pipeline = Pipeline(workspace, steps=[automl_step])
```

The pipeline is then submitted remotely for execution. However, the featurization config is totally ignored by the run: the columns are not dropped during featurization. Can anyone please check if this is a bug or tell me what I'm doing wrong here? Thank you in advance!

**Comment by justas-exadel** [comment]
I have a similar issue and filled in a [ticket](https://github.com/Azure/MachineLearningNotebooks/issues/1085)

**Comment by swatig007** [comment]
@CESARDELATORRE can you pls take a look at this issue?

**Comment by CESARDELATORRE** [comment]
@kelmahrsi Do you have a different behavior (what you expect) if running the same experiment/config with a simple AutoMLConfig out of a pipeline/AutoMLStep?

@kelmahrsi can you provide the Run ID of the run with this issue plus its log files?

Adding some folks from the team to confirm and help with the issue.
Please, @anupsms @pieths can you advise here?

**Comment by anupsms** [comment]
@pieths is looking into this

**Comment by pieths** [comment]
@kelmahrsi  Nothing sticks out initially as to why the featurization config does not work as expected with the code you posted above.

I have verified that dropping columns using `FeaturizationConfig.add_drop_columns()` does work with azureml SDK 1.13.0.

Could you try using the latest SDK and see if you still encounter the issue?

If you look at the `70_driver_log.txt` file for the setup run in the "Outputs + Logs" section for the AutoMLStep node in the UI you should see something like the following:

```console
2020-09-02 00:12:10.629 - INFO - Start updating column purposes using customized feature type settings.
2020-09-02 00:12:10.629 - INFO - Updating column number 0 to Ignore.
2020-09-02 00:12:10.629 - INFO - End updating column purposes using customized feature type settings.
```

If you look at the `70_driver_log.txt` file for the featurization run in the "Outputs + Logs" section for the AutoMLStep node in the UI you should see something like the following (note, the original dataset had 3 columns where the first was dropped):

```console
2020-09-02 00:13:59.315 - INFO - _X_train_transformed: (46, 2)
```

The `DataTransformer` in the final model should also show that the column is dropped.

```python
automl_step_run = AutoMLStepRun(pipeline_run.find_step_run("automl_train")[0])
best_run, model = automl_step_run.get_output()
summary = model.named_steps['datatransformer'].get_featurization_summary()
print(summary)
```

```console
[{'Dropped': 'Yes',
  'EngineeredFeatureCount': 0,
  'RawFeatureName': 'ColA',
  'Transformations': [''],
  'TypeDetected': 'Ignore'},
 ...]
```

**Comment by kelmahrsi** [comment]
@pieths I will try this tomorrow and report back. It might have been corrected since I was using SKD version 1.10 or 1.11 when I opened the issue.


For the record, this is what the `70_driver_log.txt` looked like when the configuration is submitted directly through the experiment from a Notebook (columns are dropped):

```
2020-07-29 18:55:30.621 - INFO - The stats_computation for raw data are following:-
	Number of Ignore features: 16
	Number of DateTime features: 1
	Number of Hashes features: 0
	Number of CategoricalHash features: 2
	Number of Categorical features: 6
	Number of AllNan features: 0
	Number of Text features: 0
	Number of Numeric features: 46
```

And here is what it looked like when submitted as an AutoMLStep in a Pipeline (configuration ignored):

```
2020-08-03 11:53:52.430 - INFO - The stats_computation for raw data are following:-
	Number of Ignore features: 0
	Number of Hashes features: 0
	Number of DateTime features: 1
	Number of CategoricalHash features: 1
	Number of Numeric features: 56
	Number of AllNan features: 0
	Number of Categorical features: 13
	Number of Text features: 0
```

**Comment by kelmahrsi** [comment]
@pieths just had the time to check! I confirm the issue is resolved as of SDK version 1.13.0.
